che-incubator/che-code
# [superseded] fix: load vs-code webview resources from local host instead of getting them from CDN

### What does this PR do?
Forwards requests to the local domain instead of sending them to the CDN.

There are four types of the requests / resources that handled:

- resources to initialize webview and webworker ( with HTTP scheme )
  `${WebviewResourcesBaseURI}oss-dev/static/out/vs/workbench/contrib/webview/browser/pre`

- resources, that are located in the current project ( FILE scheme )
  `file:///projects/web-nodejs-sample/images/default.png`

- resources from a built-in extension ( FILE scheme )
  `file:///checode/checode-linux-libc/extensions/markdown-language-features/media/pre.js`

- resources from installed extension ( FILE scheme )
  `file:///checode/remote/extensions/atlassian.atlascode-3.0.2-universal`

To be able to load resources from installed extension and from the current project it needs to create symlinks to `/project` and to `/checode/remote/extensions` directories ( see changes in assembly.Dockerfile )

![Screenshot from 2023-04-05 19-10-28](https://user-images.githubusercontent.com/1655894/230140396-36f71327-655e-411c-9d27-3643c3f57caf.png)

### What issues does this PR fix?


https://issues.redhat.com/browse/CRW-3885

### How to test this PR?

1. Create a workspace from the repository https://github.com/vitaliy-guliy/web-nodejs-sample/tree/test-webviews
  ( it will use prebuilt [quay.io/vgulyy/che-code:webviews-behind-a-proxy-stable](https://quay.io/vgulyy/che-code:webviews-behind-a-proxy-stable) editor image )

2. Open README.md at the project root, click Open Preview at the right top
3. Install Jira and Bitbucket extension, open Atlassian Settings tab ( webview based )

Before 2 and 3 steps, open the browser console, navigate to Network.
After 2 and 3 check there is no requests to vscode-cdn.net.
